### PR TITLE
SLVS-1304 Telemetry data migration to SLCore

### PR DESCRIPTION
**Pull request by vnaskos-sonar**

### CODE REVIEW

File: src/Integration/Telemetry/Migration/TelemetryDataRepository.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+[Export(typeof(ITelemetryDataRepository))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+public sealed class TelemetryDataRepository : ITelemetryDataRepository
+{
+    private readonly string storageFilePath;
+    private readonly XmlSerializer telemetrySerializer = new(typeof(TelemetryData));
+    private readonly IFileSystem fileSystem;
+
+    public TelemetryData Data { get; private set; } = new() { IsAnonymousDataShared = true };
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/Telemetry/Migration/TelemetryDataRepository.cs:38]: we should not do anything if data is not already on disk

File: src/SLCore/Configuration/SlCoreTelemetryMigrationProvider.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    [ImportingConstructor]
+    public SlCoreTelemetryMigrationProvider(ITelemetryDataRepository telemetryDataRepository)
+    {
+        this.telemetryDataRepository = telemetryDataRepository;
+    }
+
+    public TelemetryMigrationDto Get()
+    {
+        var telemetryData = telemetryDataRepository.Data;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore/Configuration/SlCoreTelemetryMigrationProvider.cs:41]: If data is null, return null

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Core/Telemetry/Migration/TelemetryData.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel;
+using System.Globalization;
+using System.Xml.Serialization;
+
+namespace SonarLint.VisualStudio.Core.Telemetry.Migration;
+
+public sealed class TelemetryData
```
[the remark this answers is not in the document]
Comment (vnaskos-sonar) [reply src/Core/Telemetry/Migration/TelemetryData.cs:27]: Changed the namespace from `Migration` to `Legacy`.

File: src/Core/Telemetry/Migration/ITelemetryDataRepository.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.Core.Telemetry.Migration;
+
+public interface ITelemetryDataRepository
```
[the remark this answers is not in the document]
Comment (vnaskos-sonar) [reply src/Core/Telemetry/Migration/ITelemetryDataRepository.cs:23]: Changed the namespace from `Migration` to `Legacy`.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**